lldb/source/API/SBSaveCoreOptions.cpp

```diff
@@ -75,6 +77,18 @@ lldb::SaveCoreStyle SBSaveCoreOptions::GetStyle() const {
   return m_opaque_up->GetStyle();
 }
 
+SBError SBSaveCoreOptions::SetProcess(lldb::SBProcess process) {
+  return m_opaque_up->SetProcess(process.GetSP());
+}
+
+SBError SBSaveCoreOptions::AddThread(lldb::SBThread thread) {
+  return m_opaque_up->AddThread(thread.get());
```

Comment:
We might want to add a `ThreadSP SBThread::get_sp()` function to avoid people possibly putting a raw "Thread *" into another shared pointer that isn't the same as the one that constructed it.

Reply:
+1. Thread also has `enable_shared_from_this`, so that is an option as well.

Reply:
Added this, and explicitly kept it lower case for the convention. Should we open an issue to look for casts to pointer and then `GetSP()` and clean them up?